storm-core/src/jvm/backtype/storm/messaging/netty/Client.java

```diff
@@ -142,6 +147,15 @@ public void run() {
                 
             }
         };
+
+        connector = new Runnable() {
+            @Override
+            public void run() {
+                if (!closing) {
+                    connect();
+                }
+            }
+        };
         
         long initialDelay = Math.min(30L * 1000, max_sleep_ms * max_retries); //max wait for 30s
```

Comment:
Math.min(arg1, arg2) method returns the smaller of the two arguments.
in this case, the maximum delay corresponds to the minimum sent by ".min" method. The comment is perhaps not accurate, but I think this is not really a problem.